files/default/test-neutron-ha-tool.py

```diff
@@ -0,0 +1,392 @@
+        picker = ha_tool.SingleAgentPicker(None)
+        picker.set_agents([])
+
+        self.assertRaises(ValueError, picker.pick)
+
+
+def configure_with(cmdline="", qclient='irrelevant'):
+    parser = ha_tool.make_argparser()
+    args = parser.parse_args(cmdline.split())
+
+    ha_tool.configure(args, qclient)
+
+
+class TestMakeConfiguration(unittest.TestCase):
+    def test_default_strategy_is_least_busy_agent_picker(self):
+        configure_with('')
+
+        self.assertIsInstance(
+            ha_tool.Configuration.agent_picker,
+            ha_tool.LeastBusyAgentPicker
+        )
+
+    def test_neutron_client_injected_to_least_busy_agent_picker(self):
+        configure_with('', 'qclient')
+
+        self.assertEqual(
+            'qclient',
+            ha_tool.Configuration.agent_picker.qclient
+        )
+
+    def test_selecting_random_agent_picker(self):
+        configure_with('--agent-selection-mode random')
+
+        self.assertIsInstance(
+            ha_tool.Configuration.agent_picker,
+            ha_tool.RandomAgentPicker
+        )
+
+    def test_explicit_agent_specified(self):
+        configure_with('--target-agent xyz')
+
+        self.assertIsInstance(
+            ha_tool.Configuration.agent_picker,
+            ha_tool.SingleAgentPicker
+        )
+        self.assertEqual(
+            'xyz',
+            ha_tool.Configuration.agent_picker.agent_selection_value
+        )
+
+
+class TestRouterFilterConfiguration(unittest.TestCase):
+    def test_null_router_filter_is_the_default(self):
+        configure_with('')
+
+        self.assertIsInstance(
+            ha_tool.Configuration.router_filter,
+            ha_tool.NullRouterFilter
+        )
+
+    def test_set_router_list(self):
+        configure_with('--router-list-file routers.lst')
+
+        self.assertIsInstance(
+            ha_tool.Configuration.router_filter,
+            ha_tool.ListFileBasedRouterFilter
+        )
+        self.assertEqual(
+            'routers.lst',
+            ha_tool.Configuration.router_filter.router_list_file
+        )
+
+
+class TestListFileBasedRouterFilter(unittest.TestCase):
+    def test_if_file_does_not_exist_system_error_raised(self):
+        router_filter = ha_tool.ListFileBasedRouterFilter('nonexisting')
+
+        with self.assertRaises(IOError) as ctx:
+            router_filter.load()
+
+    def test_loading_router_list_file(self):
+        with tempfile.NamedTemporaryFile() as list_file:
+            list_file.write('id-1\n')
+            list_file.write('\n')
+            list_file.write('     \n')
+            list_file.write('\t\t\n')
+            list_file.write('id-2')
+            list_file.seek(0)
+            
+            router_filter = ha_tool.ListFileBasedRouterFilter(list_file.name)
+            router_filter.load()
+
+        self.assertEqual(
+            ['id-1', 'id-2'],
+            router_filter.router_list
+        )
+
+    def test_filtering_empty_file(self):
```

Comment:
IMO this is a useless tests. You are just checking that the class sets the initial value to `router_list` which is not really helpful. Is the `.load()` method the one that fills the `router_list` so it doesn't make sense to check this.